pkg/operators/localmanager/localmanager.go

```diff
@@ -203,9 +205,19 @@ partsLoop:
 			}
 		}
 
+		cleanSocketPath, err := securejoin.SecureJoin(host.HostRoot, socketPath)
+		if err != nil {
+			log.Debugf("securejoin failed: %s", err)
+		}
+
+		if _, err := os.Stat(cleanSocketPath); err != nil {
+			log.Warnf("Ignoring runtime %q with non-existent socketPath %q", runtimeName, socketPath)
+			continue partsLoop
+		}
+
 		r := &containerutilsTypes.RuntimeConfig{
 			Name:            runtimeName,
-			SocketPath:      socketPath,
+			SocketPath:      cleanSocketPath,
```

Comment:
I think if we're able to get the socket path from kubernetes, we need to pass `RuntimeProtocol: containerutilsTypes.RuntimeProtocolCRI,` here, otherwise it'll try to connect to the socket using a wrong protocol:

```bash 
time="2024-05-28T17:01:43Z" level=warning msg="Skip pod kube-system/kube-controller-manager-minikube: cannot find container (ID: docker://b3ce954cfe3d7993f45a83fecf14e3c6ed090eee9f9be52f135fba10bead22e6): Get \"http://%2Fhost%2Fvar%2Frun%2Fcri-dockerd.sock/v1.45/containers/b3ce954cfe3d7993f45a83fecf14e3c6ed090eee9f9be52f135fba10bead22e6/json\": net/http: HTTP/1.x transport connection broken: malformed HTTP response \"\\x00\\x00\\x06\\x04\\x00\\x00\\x00\\x00\\x00\\x00\\x05\\x00\\x00@\\x00\".\n* Are you trying to connect to a TLS-enabled daemon without TLS?"
```